framework/contracts/account/tests/xion.rs

```diff
@@ -255,3 +255,269 @@ fn execute_from_res(deps: DepsMut, env: &Env, res: Response) -> AccountResult<Re
+                    description: None,
+                    link: None,
+                    authenticator: Some(AddAuthenticator::Secp256K1 {
+                        id: 1,
+                        pubkey: Binary::new(wallet.public_key().to_bytes()),
+                        signature: Binary::new(signature.to_vec()),
+                    }),
+                },
+                Some("Abstract Account"),
+                Some(&account_addr),
+                &[],
+                salt,
+            )
+            .map_err(Into::into)?;
+
+        let account_id = chain
+            .wasm_querier()
+            .item_query(&account_addr, state::ACCOUNT_ID)?;
+        let contract_id = format!("{ACCOUNT}-{account_id}");
+
+        let account = AccountI::new(contract_id, chain);
+        account.set_address(&account_addr);
+        Ok(account)
+    }
+
+    #[test]
+    #[serial_test::serial]
+    fn xion_account_creation() -> anyhow::Result<()> {
+        let mock = MockBech32::new("xion");
+
+        let abstr = Abstract::deploy_on(mock, ())?;
+
+        // We create an XION abstract account
+        create_xion_account(&abstr)?;
+
+        Ok(())
+    }
+
+    fn before_hook(
+        abstr: &Abstract<MockBech32>,
+        account: &AccountI<MockBech32>,
+    ) -> anyhow::Result<()> {
+        let mock = abstr.registry.environment().clone();
+        let sign_doc_bytes = Binary::from(HexBinary::from_hex("a527761bf3e9279be8cf")?);
+        let signature = xion_wallet()?.sign(&sign_doc_bytes).unwrap();
+
+        let auth_id = crate::auth_id::AuthId::new(1u8, false).unwrap();
+        let smart_contract_sig = auth_id.signature(signature.to_vec());
+        mock.app.borrow_mut().sudo(SudoMsg::Wasm(WasmSudo {
+            contract_addr: account.address()?,
+            message: to_json_binary(&AccountSudoMsg::BeforeTx {
+                msgs: vec![],
+                tx_bytes: sign_doc_bytes,
+                cred_bytes: Some(smart_contract_sig.into()),
+                simulate: false,
+            })?,
+        }))?;
+        Ok(())
+    }
+
+    fn after_hook(
+        abstr: &Abstract<MockBech32>,
+        account: &AccountI<MockBech32>,
+    ) -> anyhow::Result<()> {
+        let mock = abstr.registry.environment().clone();
+
+        mock.app.borrow_mut().sudo(SudoMsg::Wasm(WasmSudo {
+            contract_addr: account.address()?,
+            message: to_json_binary(&AccountSudoMsg::AfterTx { simulate: false })?,
+        }))?;
+        Ok(())
+    }
+
+    #[test]
+    #[serial_test::serial]
+    fn create_sub_account() -> anyhow::Result<()> {
+        let mock = MockBech32::new("xion");
+
+        let abstr = Abstract::deploy_on(mock.clone(), ())?;
+
+        // We create an XION abstract account
+        let account = create_xion_account(&abstr)?;
+
+        before_hook(&abstr, &account)?;
+        let test = mock
+            .wasm_querier()
+            .item_query(&account.address()?, AUTH_ADMIN)?;
+        println!("{:?}", test);
```

Comment:
Should be removed or changed into an assertion

Reply:
Will do